doc/user-guide/windowing.md

```diff
@@ -0,0 +1,187 @@
+ :window/aggregation :conj
+ :window/window-key :event-time
+ :window/range [30 :minutes]
+ :window/slide [5 :minutes]
+ :window/doc "Collects segments on a 30 minute window sliding every 5 minutes"}
+```
+
+#### `:count`
+
+The `:count` operation counts the number of segments in the window.
+
+```clojure
+{:window/id :count-segments
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :count
+ :window/window-key :event-time
+ :window/range [1 :hour]
+ :window/doc "Counts segments in one hour fixed windows"}
+```
+
+#### `:sum`
+
+The `:sum` operation adds the values of `:window/sum-key` for all segments in the window.
+
+```clojure
+{:window/id :sum-ages
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :sum
+ :window/sum-key :age
+ :window/window-key :event-time
+ :window/range [1 :hour]
+ :window/doc "Adds the :age key in all segments in 1 hour fixed windows"}
+```
+
+#### `:min`
+
+The `:min` operation retains the minimum value found for `:window/min-key`. An initial value must be supplied via `:window/init`.
+
+```clojure
+{:window/id :min-age
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :sum
+ :window/init 100
+ :window/min-key :age
+ :window/window-key :event-time
+ :window/range [30 :minutes]
+ :window/doc "Finds the minimum :age in 30 minute fixed windows, default is 100"}
+```
+
+#### `:max`
+
+The `:max` operation retains the maximum value found for `:window/max-key`. An initial value must be supplied via `:window/init`.
+
+```clojure
+{:window/id :max-age
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :sum
+ :window/init 0
+ :window/max-key :age
+ :window/window-key :event-time
+ :window/range [30 :minutes]
+ :window/doc "Finds the maximum :age in 30 minute fixed windows, default is 0"}
+```
+
+#### `:average`
+
+The `:average` operation maintains an average over `:window/average-key`. An initial value must be supplied via `:window/init`. The state is maintained as a map with two keys - `:n`, the number of elements, and `:average`, the running average.
+
+```clojure
+{:window/id :max-age
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :sum
+ :window/init 0
+ :window/max-key :age
+ :window/window-key :event-time
+ :window/range [30 :minutes]
+ :window/doc "Finds the maximum :age in 30 minute fixed windows, default is 0"}
+```
+
+### Window Specification
+
+See the Information Model chapter for an exact specification of what values the Window maps need to supply. Here we will describe what each of the keys mean.
+
+| key name             |description
+|----------------------|-----------
+|`:window/id`          | A unique identifier per window
+|`:window/task`        | The workflow task over which the window operates
+|`:window/type`        | Which type of window this is (fixed, sliding, etc)
+|`:window/aggregation` | The aggregation function to apply, as described above
+|`:window/window-key`  | The key over which range 
```

Comment:
Description could be better here "The key over which the range will be calculated?